scripts/update_release.py

```diff
@@ -0,0 +1,141 @@
+
+    def __init__(self, template_file=RELEASE_DEFAULT_TEMPLATE):
+        self._template = os.path.join(os.path.curdir, RESOURCES_DIR, template_file)
+        self._release = os.path.join(OUTPUT_DIR, r"release_v{V}.md")
+
+    def with_version(self, version: str):
+        if not version:
+            print("No version provided. Will use first version from changelog file")
+        self._version = version
+        return self
+
+    def with_changes(self, changes: list):
+        self._changes = self._stringfy_list(changes)
+        return self
+
+    def with_users(self, users: list):
+        self._users = self._stringfy_list(users)
+        return self
+
+    def _stringfy_list(self, items: list) -> str:
+        return "\n".join(items)
+
+    def write(self):
+        output_file = self._release.replace(r"{V}", self._version)
+        with open(output_file, "w", encoding="utf-8") as output:
+            with open(self._template, "r") as template:
+                for line in template:
+                    output.write(
+                        line.replace(r"{APP_VERSION}", self._version)
+                        .replace(r"{UPDATES}", self._changes)
+                        .replace(r"{PEOPLE}", self._users)
+                    )
+
+
+class ChangeLogParser:
+    _g_prefix_line = "-   "
+    _g_begin_changes = "<!--- Begin changes - Do not remove -->"
+    _g_end_changes = "<!--- End changes - Do not remove -->"
+    _g_begin_users = "<!--- Begin users - Do not remove -->"
+    _g_end_users = "<!--- End users - Do not remove -->"
+
+    version = ""
+    changes = []
+    users = []
+
+    def __init__(self, changelog_file: str, version: str = None):
+        self.file_path = str(changelog_file)
+        self.version = version
+
+    def __enter__(self):
+        self.file = open(self.file_path, "r")
+        return self
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        self.file.close()
+
+    def parse(self):
+        if not self.version:
+            self.parse_version()
+        self.parse_changes()
+        self.parse_users()
+
+    def parse_version(self):
+        version_regex = re.compile(r"#* *(\d+.\d+.\d+)")
+        for line in self.file:
+            if match := version_regex.match(line):
+                self.version = match.group(1)
+                return
+
+    def parse_changes(self):
+        self.changes = self.get_list_between(self._g_begin_changes, self._g_end_changes)
+
+    def parse_users(self):
+        self.users = self.get_list_between(self._g_begin_users, self._g_end_users)
+
+    def get_list_between(self, start_str: str, end_str: str):
+        in_items = False
+        items = []
+        for line in self.file:
+            if start_str in line:
+                in_items = True
+            elif in_items and end_str in line:
+                in_items = False
+                break
+            elif in_items and "-" in line[0]:
+                items.append(line.replace("\n", ""))
```

Comment:
If you use `strip`, there's no need to specify the replacement characters:

```suggestion
                items.append(line.strip())
```

Reply:
noice